src/fluree/db/query/exec/select.cljc

```diff
@@ -41,12 +41,12 @@
 (defprotocol ValueSelector
   "Format a where search solution (map of pattern matches) by extracting and
   displaying relevant pattern matches."
-  (format-value [fmt db iri-cache compact error-ch solution]))
+  (format-value [fmt db iri-cache context compact error-ch solution]))
```

Comment:
We're passing both the `context` value and the compacting function `compact` here. Was there an issue with adding the new 2-arity function to compact data maps by accepting both a context and the data value that needs compacting?

Reply:
The `json-ld/compact-fn` function take a context and optionally a "used-atom" to track which context entries are used. It then transforms the supplied context and closes over the result in order to do efficient compaction and returns a single-arity function that takes an iri.

I considered adding another arity to the returned compact-fn that would accept a context along with the iri, but that forces us to re-transform the context for every compaction, an operation that can be quite costly if the context is large. So I elected to do the more clumsy approach of passing both.

Reply:
Now that I think about it, though, if we did go with the 2-arity compact-fn it would be called in multiple places with the same context, so if we added an LRU cache of size 1 in front of it that could make the code more elegant and not incur the penalty of re-processing the context for each compaction. Or just use a delay?

Reply:
The atom isn't used for any of the compacting functions in this pull request; [it's only used for commits](https://github.com/fluree/db/blob/main/src/fluree/db/json_ld/commit.cljc#L189).

As I mentioned on our last call, I think we should define a new function that takes a context and data needed to be compacted using that context and returns the compacted data for these usages, and we should only pass the *context* through these call stacks, not any compacting functions.

Reply:
My understanding is that `compact` is called multiple times in the same call stack, and doing the work to compact is expensive, which is why we rely on the `compact-fn` to do the expensive work once.

In the the `fluree.db.query.exec.select` namespace it looks like each of the different Selectors can call `compact` to format the results, and my understanding is that several Selectors may be called during one query. If that's the case, then re-creating the `compact-fn` for each of those calls could get expensive.

But! My understanding of how Selectors work may be wrong - if only one of them is used than we would only have one compact-fn per query.